R/cal-apply.R

```diff
@@ -5,77 +5,160 @@
 #' the estimate columns names, and levels, from the calibration object.
 #' @param .data An object that can process a calibration object.
 #' @param object The calibration object (`cal_object`).
+#' @param prediction (Optional) Column identifier with the prediction.
+#' @param threshold (Optional) Applies to binary models only. A number between
+#' 0 and 1. It contains the cut off for the prediction
 #' @param ... Optional arguments; currently unused.
 #' @examples
 #' w_calibration <- cal_estimate_logistic(segment_logistic, Class)
 #'
 #' cal_apply(segment_logistic, w_calibration)
 #' @export
-cal_apply <- function(.data, object, ...) {
+cal_apply <- function(.data, object, prediction = NULL, threshold = NULL, ...) {
```

Comment:
I would remove the `threshold` argument and just assign the hard class prediction to the class with the largest probability. 

My rationale is that the function should do one thing (calibration) and not increase the scope of what this function does.

Reply:
Also, are we adding a `parameters` argument like there is `collect_predictions()`?